### ISSUE TITLE
On iOS, statusbar orientation should be used instead of device orientation

### ISSUE THREAD

**Issue by speigg** [NativeScript/NativeScript#1437]
In `application.ios.ts`, the orientation should be coming from the status bar, not the device. On iOS, `UIDevice.currentDevice().orientation` describes the orientation of the device, and is of type `UIDeviceOrientation`, while  `UIApplication.sharedApplication().statusBarOrientation` describes the orientation of the interface, and is of type `UIInterfaceOrientation` (a type which doesn't include FaceUp or FaceDown as orientations). 

The following is the line which should be changed to grab interface orientation from  `UIApplication.sharedApplication().statusBarOrientation` rather than `UIDevice.currentDevice().orientation`
https://github.com/NativeScript/NativeScript/blob/master/application/application.ios.ts#L180

**Comment by speigg** [comment]
More info on the distinction between and use-cases for UIDeviceOrientation vs UIInterfaceOrientation can be found here: https://happyteamlabs.com/blog/ios-using-uideviceorientation-to-determine-orientation/

**Comment by hshristov** [comment]
@speigg Are you seeing some issues when `UIDevice.currentDevice().orientation` is used?

**Comment by speigg** [comment]
Yes, device orientation does not map correctly to the interface orientation, i.e, faceup and facedown device orientation is mapped to "unknown" interface orientation. If the device is in one orientation the interface is not guaranteed to be in the same orientation, for any number of reasons (orientation lock may be on, certain interface orientations may not be supported by the device, etc.)

**Comment by speigg** [comment]
So basically, the device orientation really does not tell you anything about the current interface orientation. They are two distinct concepts. They should both be available via two different {N} APIs. I'm guessing the android code has similar problems here.

**Comment by x4080** [comment]
I agree with unknown thing so +1

**Comment by enchev** [comment]
Hey folks,

UIApplication.sharedApplication().statusBarOrientation is deprecated in iOS 9.0:
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIApplication_Class/#//apple_ref/occ/instp/UIApplication/statusBarOrientation

**Comment by x4080** [comment]
Oh well :)

**Comment by fealebenpae** [comment]
I believe that `-[UIApplication statusBarOrientation]` has been deprecated in favor of using `UITraitCollection` and size classes.

**Comment by speigg** [comment]
Darn, you're right, it is deprecated.

**Comment by speigg** [comment]
Actually, according to an [Apple employee](https://forums.developer.apple.com/thread/12937), "Reading the status bar orientation is not deprecated, only writing to it is. This may have been an error in how we constructed the header if you are seeing this on the getter."

**Comment by speigg** [comment]
Yeah, looking at the headers confirms it, only the setter is deprecated:
```obj-c
@property(readonly, nonatomic) UIInterfaceOrientation statusBarOrientation __TVOS_PROHIBITED;

// Explicit setting of the status bar orientation is more limited in iOS 6.0 and later.
@property(readwrite, nonatomic) UIInterfaceOrientation statusBarOrientation NS_DEPRECATED_IOS(2_0, 9_0,
"Explicit setting of the status bar orientation is more limited in iOS 6.0 and later") __TVOS_PROHIBITED;
```

**Comment by tzraikov** [comment]
@speigg, thank you for this suggestion. Indeed using statusBarOrientation is better than using currentDevice's orientation property because it will take in consideration the supported application interface orientations. However, there are some issue reports related with statusBarOrientation property when subclassing from UIWindow and when using this property at early stages of the application life-cycle. We will investigate this further and we will consider replacing the currentDevice approach with statusBarOrientation.

**Comment by vchimev** [comment]
Hey All,

Orientation property has been added to the application which is based on `UIApplication.sharedApplication.statusBarOrientation` for iOS.

It is going to be released with `6.1.0` and is already available with `tns-core-modules@next`.
